pkg/server/config.go

```diff
@@ -1,16 +1,41 @@
 package server
 
 import (
-	"encoding/json"
 	"net/http"
+
+	"gopkg.in/yaml.v2"
 )
 
+type configResponse struct {
+	Yaml string `json:"yaml"`
+}
+
 func (ctrl *Controller) configHandler(w http.ResponseWriter, _ *http.Request) {
-	configBytes, err := json.MarshalIndent(ctrl.config, "", "  ")
+	config := *ctrl.config
+	const sensitive = "<sensitive>"
+	if config.Auth.JWTSecret != "" {
+		config.Auth.JWTSecret = sensitive
+	}
+	if config.Auth.Github.ClientSecret != "" {
+		config.Auth.Github.ClientSecret = sensitive
+	}
+	if config.Auth.Google.ClientSecret != "" {
+		config.Auth.Google.ClientSecret = sensitive
+	}
+	if config.Auth.Gitlab.ClientSecret != "" {
+		config.Auth.Gitlab.ClientSecret = sensitive
+	}
+	if config.Auth.Internal.AdminUser.Password != "" {
+		config.Auth.Internal.AdminUser.Password = sensitive
+	}
+
```

Comment:
I masked everything that looked like a secret in server configuration. Have I missed anything?

Reply:
Or even better, can we somehow do it with annotations? For example, to avoid these things showing up in JSON output we add `json:"-"` annotations to fields. Maybe we can do the same thing with YAML?

Reply:
The idea was to provide a hint that the parameter has been configured but not necessarily something we should display.